spec/puppet-server-lib/puppet/jvm/compiler_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'puppet/server/compiler'
+require 'puppet/indirector/facts/mock'
+require 'puppet/indirector/catalog/mock'
+
+describe Puppet::Server::Compiler do
+  let(:request) do
+    {
+      "certname" => "foobarr",
+      "environment" => "production",
+      "persistence" => {
+        "facts" => true,
+        "catalog" => true
+      },
+      "facts" => {
+        "values" => {
+          "foo" => "bar"
+        }
+      },
+      "trusted_facts" => {
+        "values" => {
+          "foo" => "bar"
+        }
+      }
+    }
+  end
+
+
+  it 'can be instantiated' do
+    Puppet::Server::Compiler.new
+  end
+
+  it 'calls termini to save as needed' do
```

Comment:
This is a pretty sad test and just checks that our code in a few of paths doesn't blow up, but not that our inputs to the termini are correct. I don't know if we want to try to put expectations on the content of the requests given to them (and/or return a valid objects from `find`), or defer that kind of checking to acceptance tests.

Reply:
I think I'd rather test this further at a higher level. Those kinds of details in tests like this make it even harder to change any of that stuff later (e.g. if for some reason we stop using this machinery).